app/src/main/java/org/jellyfin/androidtv/ui/playback/VideoQualityController.kt

```diff
@@ -0,0 +1,22 @@
+package org.jellyfin.androidtv.ui.playback
+
+import org.jellyfin.androidtv.constant.QualityProfiles;
+import org.jellyfin.androidtv.preference.UserPreferences;
+import org.koin.java.KoinJavaComponent;
+
+class VideoQualityController(
+	previousQualitySelection: QualityProfiles
+) {
+
+	var currentQuality = previousQualitySelection
+	set(value) {
+		KoinJavaComponent.get<UserPreferences>(UserPreferences::class.java)[UserPreferences.maxBitrate] =
```

Comment:
inject user preferences via constructor to avoid KoinKavaComponent

Reply:
Done.